website/content/en/guides/level-up/e2e-ack-guide.md

```diff
@@ -0,0 +1,97 @@
+[pipeline]: /docs/setup/quickstart
+{{< /requirement >}}
+
+
+Vector is often deployed in mission critical environments where loss of data is
+non-negotiable. Ergo, assurance that every data flowing through your Vector
+pipeline is successfully delivered to the end destination is critical.
+
+We're excited to walk you through exactly you can accomplish this by using
+Vector's end-to-end acknowledgement feature.
+
+## Getting Started
+
+You can set and control the acknowledgement feature either at the global level or
+the sink level. Let's start with the global configuration option first, as it is
+easy as flipping on a switch. As you would expect from a global configuration, the
+configuration below turns end-to-end acknowledgement on for every source:
+
+```toml
+acknowledgements.enabled = true
+```
+
+Even if you have a relatively complex topology and sending data from one source to
+multiple sinks, by enabling this global config, all sources with acknowledgement
+support will wait before responding for acknowledgement from all the sinks that
+its events flow through.
+
+If any sink rejects a message, that failure marks the batch that the message is
+part of as _failed_, and the source will respond accordingly. For example, HTTP sources
+will produce some 400 error code, while sources like Kafka that don't handle
+rejection behaves similarly to a positive acknowledgement.
+
+But you can enable acknowledgements individually for each sink if you want more granular
+control over how acknowledgements for specific cases (e.g. you're buffering your
+data). When you enable `acknowledgements` for a supported sink, all sources connected
+to that sink will wait for delivery. You can set acknowledgements at the sink level by
+doing the following:
+
+```toml
+[sinks.very_cool_id]
+   # Enable or disable waiting for acknowledgements for this sink.
+   # Defaults to the global value of `acknowledgements`
+   acknowledgements.enabled = true
+```
+
+## Edge cases
+
+Unsurprisingly, there are a few exceptions and edge cases for the end-to-end
+acknowledgement feature.
+
+First, as alluded to earlier, not all sources and sinks are supported. We are, of
+course, working to add support for more sources and sinks, but support for some
+sources and sinks will always be unable because they cannot provide acknowledgements
+at the protocol level.
+
+Sources that don't support acknowledgements will output a
+warning message to let you know if you try to enable acknowledgements, while sinks
+that don't support acknowledgements will trigger a fatal error and prevent the
+configuration from loading. You can find which components support acknowledgements
+on each of their documentation pages by looking for the acknowledgments badge.
+
+Second, when buffering your observability data in a disk, the behavior of end-to-end
+acknowledgement is a little different. When an event is persisted in a disk buffer,
+the event will be marked as acknowledged to the source.
```

Comment:
Is it worth giving a reason why we do this? ie these events will continue be retried in the event of a crash, so we have accepted responsibility, etc.